### ISSUE TITLE
Engine: lazy-import heavy ML modules to drop cold-start from 25s to ~3s

### ISSUE THREAD

**Issue by dtubb** [dtubb/fichero#743]
## Problem

Cold-start of the briefcase-bundled engine takes ~25 seconds on M-series Macs. Captured timestamps from a real launch on 2026-04-30:

```
10:12:23.386  Starting Fichero Backend
10:12:24.380  Auth token initialized
10:12:47.886  FICHERO_FEATURE_TIER resolved          ← +23s of imports
10:12:48.305  Application startup complete
10:12:48.305  Uvicorn running on http://127.0.0.1:8765
```

The 23-second gap is dominated by eager imports of heavy ML libraries at module-load time (`langchain`, `langgraph`, `lancedb`, `torch`, `fastembed`, `kreuzberg`, `litellm`, etc.). Each `.so` from those packages triggers a code-signature validation on first load when launched from a code-signed `.app` bundle, which compounds the cost.

The Swift host had a 30-second `waitForBackend` timeout that occasionally fired right as uvicorn was about to bind. Bumped to 90s as a 0.0.2 hotfix, but the underlying problem is the cold start itself — every Fichero launch makes the user stare at "Connecting to backend…" for half a minute.

## Goal

Cold-start time from process exec to uvicorn bound: **<5 seconds** on M-series Apple Silicon.

## Approach: lazy imports inside route handlers

The principle: at module-import time, import only what FastAPI needs to *register* the routes. Defer importing the actual ML libraries until a route handler that uses them is *called*.

```python
# BEFORE — top of routes/workflows.py:
from langgraph.graph import StateGraph
from fichero.workflows.executor import run_workflow

@router.post("/api/workflows/run")
async def run(req: RunRequest):
    return await run_workflow(req)


# AFTER:
@router.post("/api/workflows/run")
async def run(req: RunRequest):
    # Heavy imports lazy: only paid when this route is actually called.
    from fichero.workflows.executor import run_workflow
    return await run_workflow(req)
```

For modules that are *always* needed at startup (DB manager, FastAPI app construction, route registration, our auth shim), keep the eager imports.

## Scope of files affected

Likely culprits at module top-level (audit needed):

- `fichero-engine/src/fichero/api/routes/*.py` — many routes pull in langchain, langgraph, embedders at import time
- `fichero-engine/src/fichero/workflows/*.py` — workflow executor and tools all eagerly import `langchain`, `langgraph`
- `fichero-engine/src/fichero/llm.py` — pulls in `litellm` at import
- `fichero-engine/src/fichero/db_embeddings.py` — pulls in `lancedb`, `fastembed`
- `fichero-engine/src/fichero/embeddings/*.py` — same

Audit step: run `python -X importtime -c "from fichero.api import main"` on the bundled engine and inspect which imports dominate. Target the top 5-10 import-time hogs.

## Things to NOT lazy-import

- `fastapi`, `uvicorn` — needed at startup, fast anyway
- `pydantic` — needed for route signatures
- `duckdb` — fast, needed for the lifespan startup that opens the app DB
- `fichero.api.auth` (our shim) — must run at startup to write the token file before uvicorn binds
- Anything used in route *type annotations* (FastAPI introspects these at registration time → can't be lazy)

## Pitfalls

1. **Type annotations.** If a route handler signature uses `langchain.schema.HumanMessage` as a type, FastAPI evaluates that at registration time → eager import. Workaround: use string annotations (`from __future__ import annotations`) or wrap in `if TYPE_CHECKING:`.
2. **Pydantic models that reference ML types.** Same problem — Pydantic introspects the model at class definition. Avoid mixing ML SDK types into request/response models; use plain dicts or your own thin types.
3. **Module-level singletons.** Some libraries (looking at `fastembed`) want to be initialized once; lazy import means first request pays the model-load cost. Mitigate with a startup background task that warms them after uvicorn binds.
4. **Workflow registration at startup.** If we register workflow tools by importing all of `fichero.workflows.tools.*` at startup so they self-register, we'd negate the savings. Move tool registration to first-workflow-run, OR keep a thin tool-name → module-path registry and import the implementation only when invoked.

## Proposed implementation sequence

1. **Audit (30 min):** run `-X importtime` on the running engine, identify the top imports by duration. Save the report.
2. **Lazy-import workflow tools (2 hours):** the biggest win. Most routes don't touch workflow tools at all; only `/api/workflows/run`-shaped endpoints do. Move tool imports inside the run handler.
3. **Lazy-import LLM/embeddings (1 hour):** `llm.py` and `db_embeddings.py` — push `litellm`, `fastembed`, `lancedb` imports into the first call site.
4. **Lazy-import LangGraph (1 hour):** the graph executor.
5. **Background warmup (30 min):** add a FastAPI startup task that imports the heavy modules in a thread *after* uvicorn binds, so the first real request doesn't pay the full cost. Best of both worlds.
6. **Measure (30 min):** re-run the timestamped boot. Target: bound port within 5 seconds.

## Acceptance

- `time "/path/to/Fichero Engine"` prints uvicorn-bound message within 5 seconds on M-series, cold cache.
- All existing tests pass (`pytest fichero-engine/tests/unit/`).
- Workflow runs still complete (no regression in correctness — just slower first-call).
- Optional: after 60s of warmup, first workflow run is no slower than before this change (background warmup hides the delay).

## Why not part of 0.0.2

- Touches many files; high regression risk for a same-day release.
- 90s timeout (committed as 0.0.2 hotfix) is sufficient for users who can wait 25s once.
- Better to ship 0.0.2 with the working-but-slow engine and tackle this with proper TDD in 0.0.3.

## Related

- Hotfix that's already in 0.0.2: bumped `EmbeddedBackendService.waitForBackend(timeout:)` from 30s → 90s.
- 0.0.3 milestone: KG Navigation + Polish (this fits there as a quality-of-life polish item).

**Comment by dtubb** [comment]
🤖 Worker: moved top-level `import torch` / `import pykeen` in `pykeen_inference.py` and `kg_predictions.py` to function-level lazy imports. The pykeen compat shim is now deferred via `_ensure_pykeen_compat()` called at first route use. Linting and test suite pass.
